googlesource/source_intergration_test.go

```diff
@@ -107,92 +104,372 @@ func dataSetup() (err error) {
 		return fmt.Errorf("job completed with error: %v", status.Err())
 	}
 
-	fmt.Println("Table created:", tableID2)
+	t.Log("Table created:", tableID2)
 
 	return nil
 }
 
-func cleanupDataSet() (err error) {
+// Item represents a row item.
+type Item struct {
+	Name      string
+	Age       int
+	Updatedat time.Time
+}
+
+// Initial setup required - project with service account.
+func dataSetupWithTimestamp(t *testing.T) (err error) {
 	ctx := context.Background()
 
-	client, err := bigquery.NewClient(ctx, projectID, option.WithCredentialsFile(serviceAccount))
+	client, err := bigquery.NewClient(ctx, projectID, option.WithCredentialsJSON([]byte(serviceAccount)))
 	if err != nil {
 		return fmt.Errorf("bigquery.NewClient: %v", err)
 	}
 	defer client.Close()
 
-	table := client.Dataset(datasetID).Table(tableID)
-	if err := table.Delete(ctx); err != nil {
+	meta := &bigquery.DatasetMetadata{
+		Location: location, // See https://cloud.google.com/bigquery/docs/locations
+	}
+
+	// create dataset
+	if err := client.Dataset(datasetID).Create(ctx, meta); err != nil && !strings.Contains(err.Error(), "duplicate") {
+		t.Log("Dataset could not be created created")
 		return err
 	}
 
-	table = client.Dataset(datasetID).Table(tableID2)
-	if err := table.Delete(ctx); err != nil {
+	sampleSchema := bigquery.Schema{
+		{Name: "name", Type: bigquery.StringFieldType},
+		{Name: "age", Type: bigquery.IntegerFieldType},
+		{Name: "updatedat", Type: bigquery.TimestampFieldType},
+	}
+
+	metaData := &bigquery.TableMetadata{
+		Schema:         sampleSchema,
+		ExpirationTime: time.Now().AddDate(1, 0, 0), // Table will be automatically deleted in 1 year.
+	}
+	tableRef := client.Dataset(datasetID).Table(tableIDTimeStamp)
+	err = tableRef.Create(ctx, metaData)
+	if err != nil && !strings.Contains(err.Error(), "duplicate") {
+		t.Log("Error: ", err)
 		return err
 	}
 
-	client, err = bigquery.NewClient(ctx, projectID, option.WithCredentialsFile(serviceAccount))
+	var query string
+
+	for i := 0; i < 8; i++ {
+		iString := fmt.Sprintf("%d", i)
+		query = "INSERT INTO `" + projectID + "." + datasetID + "." + tableIDTimeStamp + "`  values ('name" + iString +
+			"', " + iString + ", '" + time.Now().UTC().AddDate(0, 0, -i).Format("2006-01-02 15:04:05.999999 MST") + "' )"
+
+		q := client.Query(query)
+		q.Location = location
+
+		job, err := q.Run(ctx)
+		if err != nil {
+			t.Log("Error found: ", err)
+		}
+
+		status, err := job.Wait(ctx)
+		if err != nil {
+			t.Log("Error found: ", err)
+		}
+
+		if err := status.Err(); err != nil {
+			t.Log("Error found: ", err)
+		}
```

Comment:
Is it intentional that these do not return errors, but only log them?

Reply:
This is fixed in https://github.com/neha-Gupta1/conduit-connector-bigquery/pull/22 Once its merged to this branch we will start seeing that changes